instrumentation/opentelemetry-instrumentation-openai-v2/tests/test_chat_completions.py

```diff
@@ -102,56 +356,385 @@ def test_chat_completion_streaming(exporter, openai_client):
+    ]
+
+    response_0 = openai_client.chat.completions.create(
+        messages=messages_value,
+        model=llm_model_value,
+        n=2,
+        stream=True,
+        stream_options={"include_usage": True},
+    )
+
+    # two strings for each choice
+    response_stream_result = ["", ""]
+    finish_reasons = ["", ""]
+    for chunk in response_0:
+        if chunk.choices:
+            for choice in chunk.choices:
+                response_stream_result[choice.index] += (
+                    choice.delta.content or ""
+                )
+                if choice.finish_reason:
+                    finish_reasons[choice.index] = choice.finish_reason
+
+        # get the last chunk
+        if getattr(chunk, "usage", None):
+            response_stream_usage = chunk.usage
+            response_stream_model = chunk.model
+            response_stream_id = chunk.id
+
+    # sanity check
+    assert "stop" == finish_reasons[0]
+
+    spans = span_exporter.get_finished_spans()
+    assert_all_attributes(
+        spans[0],
+        llm_model_value,
+        response_stream_id,
+        response_stream_model,
+        response_stream_usage.prompt_tokens,
+        response_stream_usage.completion_tokens,
+    )
+
+    logs = log_exporter.get_finished_logs()
+    assert len(logs) == 4
+
+    system_message = {"content": messages_value[0]["content"]}
+    assert_message_in_logs(
+        logs[0], "gen_ai.system.message", system_message, spans[0]
+    )
+
+    user_message = {
+        "content": "What's the weather in Seattle and San Francisco today?"
+    }
+    assert_message_in_logs(
+        logs[1], "gen_ai.user.message", user_message, spans[0]
+    )
+
+    choice_event_0 = {
+        "index": 0,
+        "finish_reason": "stop",
+        "message": {
+            "role": "assistant",
+            "content": "".join(response_stream_result[0]),
+        },
+    }
+    assert_message_in_logs(logs[2], "gen_ai.choice", choice_event_0, spans[0])
+
+    choice_event_1 = {
+        "index": 1,
+        "finish_reason": "stop",
+        "message": {
+            "role": "assistant",
+            "content": "".join(response_stream_result[1]),
+        },
+    }
+    assert_message_in_logs(logs[3], "gen_ai.choice", choice_event_1, spans[0])
+
+
+@pytest.mark.vcr()
+def test_chat_completion_multiple_tools_streaming(
+    span_exporter, log_exporter, openai_client
+):
+    llm_model_value = "gpt-4o-mini"
+    messages_value = [
+        {"role": "system", "content": "You're a helpful assistant."},
+        {
+            "role": "user",
+            "content": "What's the weather in Seattle and San Francisco today?",
+        },
+    ]
+
+    response = openai_client.chat.completions.create(
+        messages=messages_value,
+        model=llm_model_value,
+        tool_choice="auto",
+        tools=[get_current_weather_tool_definition()],
+        stream=True,
+        stream_options={"include_usage": True},
+    )
+
+    finish_reason = None
```

Comment:
Since these are recorded shouldn't we just hardcode the numbers later instead? this way we are copying the implementation that may hide any issue.

Reply:
if we hardcode everything, then it'd become a problem if we ever need to re-record. From my experience working with test recordings the need comes every once in a while.